### DISCUSSION TITLE
Hotrod client puIfAbsent always return null

### DISCUSSION THREAD

**Discussion by davidmestr** [infinispan/infinispan#11140] [Q&A] [answered]
Hi!

We are checking the response of the putIfAbsent, and we always are getting null. We think that we should get the previous state of the cache, in order words, a null the first time (when there is no entry of this key in the cache) and the previous cached values in the following calls.

`[HotRod-client-itx-event-loop-group-1] TRACE org.infinispan.client.hotrod.impl.operations.PutIfAbsentOperation - Returning from putIfAbsent: null`

We are using SpringCache API, but it delegates in Hotrod client.

Is this the expected behaviour?

**Comment by tristantarrant** [comment, marked as answer]
Please read https://infinispan.org/docs/stable/titles/hotrod_java/hotrod_java.html#forcing-return-values_hotrod-client-configuration
